src/Kovan-DssSpell.t.sol

```diff
@@ -772,85 +833,92 @@ contract DssSpellTest is DSTest, DSMath {
-
-        // Convert 20 USDC to DAI with a 0.1% fee (lerp is done)
-        hevm.warp(now + 4 days);
-        lerpUSDCPSM.tick();
-        assertTrue(lerpUSDCPSM.done());
-        assertEq(psmUSDCPSM.wards(address(lerpUSDCPSM)), 0);    // Lerp de-auths itself
-        assertEq(psmUSDCPSM.tin(), WAD * 1 / 1000);
-        assertEq(psmUSDCPSM.tout(), WAD * 1 / 1000);
-        usdc.approve(address(joinUSDCPSM), faucetAmount);
-        psmUSDCPSM.sellGem(address(this), faucetAmount);
-        faucetAmount = faucetAmount * 999 / 1000;
-        faucetAmountWad = faucetAmount * (10 ** (18 - usdc.decimals()));
-        assertEq(usdc.balanceOf(address(this)), 0);
-        assertEq(dai.balanceOf(address(this)), faucetAmountWad);
+        assertEq(lpJoin.wards(pauseProxy), 1);
+        assertEq(vat.wards(address(lpJoin)), 1);
+        assertEq(lpFlip.wards(address(end)), 1);
+        assertEq(lpFlip.wards(address(flipMom)), 1);
+        assertEq(lpPip.wards(address(osmMom)), 1);
+        assertEq(lpPip.bud(address(spot)), 1);
+        assertEq(lpPip.bud(address(end)), 1);
+        assertEq(MedianAbstract(lpPip.orb1()).bud(address(lpPip)), 1);
+
+        // Join to adapter
+        uint256 amount = 0.01 ether;
+        hevm.store(
+            address(lp),
+            keccak256(abi.encode(address(this), uint256(1))),
+            bytes32(amount)
+        );
+        assertEq(lp.balanceOf(address(this)), amount);
+        assertEq(vat.gem(ilk, address(this)), 0);
+        lp.approve(address(lpJoin), amount);
+        lpJoin.join(address(this), amount);
+        assertEq(lp.balanceOf(address(this)), 0);
+        assertEq(vat.gem(ilk, address(this)), amount);
+
+        // Deposit collateral, generate DAI
+        assertEq(vat.dai(address(this)), 0);
+        vat.frob(ilk, address(this), address(this), address(this), int(amount), int(100 * WAD));
+        assertEq(vat.gem(ilk, address(this)), 0);
+        assertEq(vat.dai(address(this)), 100 * RAD);
+
+        // Payback DAI, withdraw collateral
+        vat.frob(ilk, address(this), address(this), address(this), -int(amount), -int(100 * WAD));
+        assertEq(vat.gem(ilk, address(this)), amount);
+        assertEq(vat.dai(address(this)), 0);
+
+        // Withdraw from adapter
+        lpJoin.exit(address(this), amount);
+        assertEq(lp.balanceOf(address(this)), amount);
+        assertEq(vat.gem(ilk, address(this)), 0);
+
+        // Generate new DAI to force a liquidation
+        lp.approve(address(lpJoin), amount);
+        lpJoin.join(address(this), amount);
+        (,,uint256 spotV,,) = vat.ilks(ilk);
+        // dart max amount of DAI
+        vat.frob(ilk, address(this), address(this), address(this), int(amount), int(mul(amount, spotV) / RAY));
+        hevm.warp(now + 1);
+        jug.drip(ilk);
+        assertEq(lpFlip.kicks(), 0);
+        cat.bite(ilk, address(this));
+        assertEq(lpFlip.kicks(), 1);
     }
 
+    function testCastCost() public {
```

Comment:
This test can be commented and just enable it when we have doubts that the spell might not be able to be cast.

Reply:
In this case it is a regular spell which won't be a problem.